ant/easy/node.py

```diff
@@ -114,8 +117,60 @@ def _worker(self):
         self.ant.start()
 
     def _main(self):
+        _logger.debug("node._main started")
+
         while self._running:
+            # SDS TODO : hier de self._events opkuisen/afhandelen
+            # dan heeft die timeout op datas.get() nog zin
+            # beter om dit per channel te doen ipv de volledige node?
+            # SDS : voorlopige workaround
+            # remove RX_FAILED, COLLISION events that keep piling up
+            # TODO : nog niet ok, want als het [0] niet wordt gecleaned of gebruikt, stapelen de vuile events zich erachter op
+            # hier moet een algo komen die systematisch alle events afhandelt!
+            _logger.debug("SDS - cleaning node._events queue %r", self._events)
+
+            # sds de .acquire() is niet echt nodig als je events.remove(message) gebruikt ipv popleft
+            # zonder de .acquire() zou de node._events queue kunnen wijzigen tussen message=... en node._events.popleft()
+            # om dezelfde reden kan je ook geen for message in node._events loop doen zonder lock acquire
+            # deze cleaning vermijdt nog niet dat wait_for_event wordt genotified bij elke inkomend event (ook die van <> channel)
+            # want deze cleaning routine krijgt niet noodzakelijk eerst toegang tot de gewijzigde node._events
+            # dit is sowieso tijdelijk, maar eigenlijk kan je ook beter een lijst hebben van events die moeten blijven ipv op te sommen wat weg moet
+            # ofwel moeten we hier die error events callbacken naar de applicatie
```

Comment:
This is also a bit unclear to me.

Reply:
I noticed that the node._events queue grows continually during execution, because the events are never cleaned. I'm not an expert in ANT, but I would expect that RX_FAILED and COLLISION events are totally normal when multiple channels are open. Every channel has a fixed channel_period; even if they're slightly different, at some point in time they will overlap and raise a COLLISION or RX_FAILED event. Same with QUE_OVERFLOW and other error events, what do you want to do with these?
And then the TODO says that the current quick hack algorithm is flawed, because if the [0] event cannot be handled, the unhandled events will pile up after it, and the cleaning doesn't work. Like I warned you, it's work in progress  ... My first goal was to keep the code running. It looks ugly, I admit